src/python/pants/backend/docker/subsystems/docker_options.py

```diff
@@ -86,16 +92,33 @@ def register_options(cls, register):
             default=[],
             advanced=True,
             help=(
-                "Environment variables to set for `docker` invocations. "
+                "Environment variables to set for `docker` invocations.\n\n"
                 "Entries are either strings in the form `ENV_VAR=value` to set an explicit value; "
                 "or just `ENV_VAR` to copy the value from Pants's own environment."
             ),
         )
 
+        register(
+            "--run-args",
+            type=list,
+            member_type=str,
+            default=[],
```

Comment:
Should set `passthrough=True`

Reply:
Are you sure? Since that flag is already set on `--run-args`.. which option will get the extra args?
```
./pants run Dockerfile -- pass through args
```

Or, perhaps, you could enlighten me as to what `passthrough=True` implies, in case it is not entirely correct and clear in my mind? :)

Reply:
`passthrough=True` causes options after `--` on Pants' CLI to be appended to the option that it is set on. So setting this option to `passthrough=True` and then running:
```console
./pants run --docker-run-args='a b c' $docker -- d e f
```
... will result in a `[docker].run_args` value of `a b c d e f`.

Reply:
But what about `--run-args` then... this is confusing me :P 
Feels like you can't have more than one option with `passthrough=True`...

Reply:
There is no `--run-args` option: there are only per-subsystem options.

Reply:
Oh, whoops. Yes, correct: I was thinking of that option.

So I don't think that this option needs to exist then? If it does need to exist, then adding some explanation of how it differs from `--run-args` would be good. Sorry about that.

Reply:
Well, here I think that the current `--run-args` as passthrough makes sense (and I'd like it to stay as-is), while the `--docker-run-args` will likely be more often used as cli option than in the pants.toml (guessing) (or a combination) as it may be very image specific what you want to put in there.

So, it may look like this:

```
./pants run --docker-run-args="--interactive --name demo --volume /mnt/images:/image-data" src/example:demo-image -- my args to my image entrypoint
```

Here, I don't want to change the way the args to the image entrypoint is done.